cpp/src/plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc

```diff
@@ -0,0 +1,314 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.h"
+
+#include <pthread.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "plasma/client.h"
+
+const jsize LEN_OF_OBJECTID = sizeof(plasma::ObjectID) / sizeof(jbyte);
+
+inline void jbyteArray_to_object_id(JNIEnv* env, jbyteArray a, plasma::ObjectID* oid) {
+  env->GetByteArrayRegion(a, 0, LEN_OF_OBJECTID, reinterpret_cast<jbyte*>(oid));
+}
+
+inline void object_id_to_jbyteArray(JNIEnv* env, jbyteArray a, plasma::ObjectID* oid) {
+  env->SetByteArrayRegion(a, 0, LEN_OF_OBJECTID, reinterpret_cast<jbyte*>(oid));
+}
+
+class JByteArrayGetter {
+ private:
+  JNIEnv* _env;
+  jbyteArray _a;
+  jbyte* bp;
+
+ public:
+  JByteArrayGetter(JNIEnv* env, jbyteArray a, jbyte** out) {
+    _env = env;
+    _a = a;
+
+    bp = _env->GetByteArrayElements(_a, NULL);
+    *out = bp;
+  }
+
+  ~JByteArrayGetter() { _env->ReleaseByteArrayElements(_a, bp, 0); }
+};
+
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_connect(
+    JNIEnv* env, jclass cls, jstring store_socket_name, jstring manager_socket_name,
+    jint release_delay) {
+  const char* s_name = env->GetStringUTFChars(store_socket_name, NULL);
+  const char* m_name = env->GetStringUTFChars(manager_socket_name, NULL);
+
+  plasma::PlasmaClient* client = new plasma::PlasmaClient();
+  ARROW_CHECK_OK(client->Connect(s_name, m_name, release_delay));
+
+  env->ReleaseStringUTFChars(store_socket_name, s_name);
+  env->ReleaseStringUTFChars(manager_socket_name, m_name);
+  return reinterpret_cast<int64_t>(client);
+}
+
+JNIEXPORT void JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_disconnect(
+    JNIEnv* env, jclass cls, jlong conn) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+
+  ARROW_CHECK_OK(client->Disconnect());
+  delete client;
+  return;
+}
+
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_create(
+    JNIEnv* env, jclass cls, jlong conn, jbyteArray object_id, jint size,
+    jbyteArray metadata) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
+  plasma::ObjectID oid;
+  jbyteArray_to_object_id(env, object_id, &oid);
+
+  // prepare metadata buffer
+  uint8_t* md = NULL;
+  jsize md_size = 0;
+  std::unique_ptr<JByteArrayGetter> md_getter;
+  if (metadata != NULL) {
+    md_size = env->GetArrayLength(metadata);
+  }
+  if (md_size > 0) {
+    md_getter.reset(new JByteArrayGetter(env, metadata, reinterpret_cast<jbyte**>(&md)));
+  }
+
+  std::shared_ptr<Buffer> data;
+  Status s = client->Create(oid, size, md, md_size, &data);
+  if (s.IsPlasmaObjectExists()) {
+    jclass Exception = env->FindClass("org/ray/spi/impl/PlasmaObjectExistsException");
```

Comment:
This class needs to be integrated into this PR, right? Can we also add a test that this works?

Reply:
yeah, the test case current add to the ci will test this file too, so the current ci is passed means this class is work as well.

Reply:
I know what you mean now, i will use the normal Exception(java.lang.Exception) instead, thanks.